examples/CMakeLists.txt

```diff
@@ -56,13 +56,24 @@ if (PTHREAD_LIBRARIES
     AND CDK_LIBRARY
 )
     find_path(LIBCKD_INCLUDE_DIR cdk.h PATH_SUFFIXES cdk)
-    include_directories(${LIBCKD_INCLUDE_DIR})
-    project(iio-monitor C)
-    add_executable(iio-monitor iio-monitor.c)
-    target_link_libraries(
-        iio-monitor iio ${PTHREAD_LIBRARIES} ${CURSES_LIBRARY} ${CDK_LIBRARY}
-    )
-    set(IIO_TESTS_TARGETS ${IIO_TESTS_TARGETS} iio-monitor)
+
+    set(TEMP ${CMAKE_REQUIRED_LIBRARIES})
+    set(TEMP1 ${CMAKE_REQUIRED_INCLUDES})
+    list(APPEND CMAKE_REQUIRED_LIBRARIES ${CURSES_LIBRARY} ${CDK_LIBRARY})
+    list(APPEND CMAKE_REQUIRED_INCLUDES ${LIBCKD_INCLUDE_DIR})
+    check_symbol_exists(CDK_CSTRING2 "cdk.h" HAS_CDK_CSTRING2)
+    set(CMAKE_REQUIRED_LIBRARIES ${TEMP})
+    set(CMAKE_REQUIRED_INCLUDES ${TEMP1})
```

Comment:
So, there's a possibility that cdk.h exists but at the same time not include symbol CDK_CSTRING2?
If yes, then cmake should print a message with the reason why iio-monitor will not be included in the build.

Reply:
On old versions of CDK - it doesn't do it. I can add a message there.